_docs/checker-workflows.md

```diff
@@ -0,0 +1,157 @@
+---
+title: Checker Workflows
+permalink: /docs/publisher-tutorials/checker-workflows/
+---
+
+# Checker Workflows
+
+## Tutorial Goals
+* Learn about checker workflows
+* Add a checker workflow to an existing tool
+* Update a checker workflow
+* Launch a checker workflow
+
+## Overview Checker Workflows
+Checker workflows are additional workflows you can associate with a tool or workflow. The purpose of them is to ensure that a tool or workflow, given some inputs, produces the expected outputs. Below is a visual overview of how a checker workflow looks.
+
+![Checker Workflow Layout](/assets/images/docs/checker-workflow.png)
+
+The term "entry" will be used as a generic term to refer to both tools and workflows.
+
+The entry that a checker workflow is testing will be referred to as an original tool/workflow/entry.
+
+For this tutorial we will be registering a checker workflow to test an original tool that calculates the MD5sum of a file.
+
+The relevant tools and workflows can be found in the following Git repository:
+[https://github.com/dockstore-testing/md5sum-checker](https://github.com/dockstore-testing/md5sum-checker)
+
+#### Quick overview of structure
+One point of confusion is that a checker workflow contains a validation tool/workflow. The validation tool/workflow is what does the bulk of the validation. It is responsible for ensuring that the results of the original entry match expected results. The checker workflow refers to the workflow that connects the original entry with the validation tool/workflow, so that it can be run as one workflow.
+
+Also like regular workflows, a checker workflow can take input from an input parameter file. The checker workflow can either use the input parameter file for the original entry, or it can define its own. The second case is useful when the validation tool/workflow has some extra parameters not required by the original entry.
+
+For our example the second case is used. The parent tool has the input parameter file [/md5sum-input-cwl.json](https://github.com/dockstore-testing/md5sum-checker/blob/master/md5sum/md5sum-input-cwl.json)
+```
+{
+  "input_file": {
+        "class": "File",
+        "path": "md5sum.input"
+    }
+}
+```
+
+The checker workflow has the input parameter file [/checker-input-cwl.json](https://github.com/dockstore-testing/md5sum-checker/blob/master/checker-input-cwl.json)
+```
+{
+  "input_file": {
+        "class": "File",
+        "path": "md5sum.input"
+    },
+    "expected_md5": "00579a00e3e7fa0674428ac7049423e2"
+}
+```
+
+Notice that the checker parameter file has the same content as the parent parameter file, in addition to having a checker specific parameter.
+
+#### Output of checker workflow
+To ensure that checking a checker workflow's output can be automated, it is important that the checker workflow produce consistent exit codes. We recommend using an exit code of 0 for success and an exit code of not 0 for failures.
```

Comment:
Maybe "require" instead of "recommend"
Exit codes are/will be used by both the challenge and the testbed